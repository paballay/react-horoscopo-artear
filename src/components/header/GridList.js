export const GridList = ( { handleList, handleGrid } ) => {

    return (
        <div className="btn-group col-md-2" role="group">
            <button type="button" className="btn btn-outline-primary m-2" onClick={ handleGrid } >Grilla</button>
            <button type="button" className="btn btn-outline-primary m-2" onClick={ handleList } >Lista</button>
        </div> 
    )
}
