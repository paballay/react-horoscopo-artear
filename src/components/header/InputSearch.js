import { useState } from "react";

export const InputSearch = ({ setInputSearch }) => {

    const [input, setInput] = useState('');

    const handleInput = ({ target: { value }}) => {
        setInput(value);

        if(value === '') setInputSearch('');
    }

    const submit = (e) =>{
        e.preventDefault();

        setInputSearch(input);
    }

    return (
        <form onSubmit={ submit } className="form-group has-warning">
            <input id="inputSearch" className="form-control form-control-warning m-2" 
                type="text" placeholder="Buscar Signo" name="signo" onChange={ handleInput } autoComplete="off"/>
        </form>
    )
}
