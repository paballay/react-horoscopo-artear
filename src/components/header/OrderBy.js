export const OrderBy = ({ handleOrderByDate, handleOrderByAlphabet }) => {
    return (
        <div className="row justify-content-start">
            <label className="col-md-4 m-3 p-0 text-end">Ordenar por: </label>
            <div className="col-md-7 p-0">
                <div className="btn-group" role="group">
                    <button type="button" className="btn btn-outline-primary m-2" onClick={handleOrderByDate} >Fecha</button>
                    <button type="button" className="btn btn-outline-primary m-2" onClick={handleOrderByAlphabet} >Nombre</button>
                </div>
            </div>

        </div>
    )
}
