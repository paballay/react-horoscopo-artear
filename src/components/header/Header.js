import { GridList } from "./GridList"
import { InputSearch } from "./InputSearch"
import { OrderBy } from "./OrderBy"

export const Header = ( { handleList, handleGrid, handleOrderByDate, handleOrderByAlphabet, setInputSearch } ) => {
    
    return (
        <>
            <h1> Hor√≥scopo </h1>
            <div className="row">
                <div className="col-md-4 p-0">
                    <GridList handleList={ handleList } handleGrid={ handleGrid }/>
                </div>
                <div className="col-md-4">
                    <OrderBy handleOrderByDate={ handleOrderByDate } handleOrderByAlphabet={ handleOrderByAlphabet }/>
                </div>
                <div className="col-md-4"> 
                    <InputSearch setInputSearch={ setInputSearch }/>                    
                </div>
            </div>
        </>
    )
}
