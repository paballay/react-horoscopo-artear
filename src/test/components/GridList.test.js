import { shallow } from 'enzyme';
import { GridList } from '../../components/header/GridList';

describe('Probando componente <GridList />', () => {
    const handleList = () => {
        console.log('Boton Lista presionado');
    }

    const handleGrid = () => {
        console.log('Boton Grilla presionado');
    }
    
    let wrapper = shallow( <GridList handleList={ handleList } handleGrid={ handleGrid }/> );

    beforeEach(() => {
        wrapper = shallow( <GridList handleList={ handleList } handleGrid={ handleGrid }/> );
    })

    test('Renderizando del componente', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Simulando boton Grilla', () => {
        wrapper.find('button').at(0).simulate('click');
    });

    test('Simulando boton Lista', () => {
        wrapper.find('button').at(1).simulate('click');
    });
})