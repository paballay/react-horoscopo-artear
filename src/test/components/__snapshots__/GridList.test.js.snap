// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente <GridList /> Renderizando componente 1`] = `
<div
  className="btn-group col-md-2"
  role="group"
>
  <button
    className="btn btn-outline-primary m-2"
    onClick={[Function]}
    type="button"
  >
    Grilla
  </button>
  <button
    className="btn btn-outline-primary m-2"
    onClick={[Function]}
    type="button"
  >
    Lista
  </button>
</div>
`;

exports[`Probando componente <GridList /> Renderizando del componente 1`] = `
<div
  className="btn-group col-md-2"
  role="group"
>
  <button
    className="btn btn-outline-primary m-2"
    onClick={[Function]}
    type="button"
  >
    Grilla
  </button>
  <button
    className="btn btn-outline-primary m-2"
    onClick={[Function]}
    type="button"
  >
    Lista
  </button>
</div>
`;
